kornia/augmentation/container/utils.py

```diff
@@ -0,0 +1,222 @@
+from typing import Callable, cast, Dict, Iterator, List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+
+import kornia.augmentation.container as CTN  # lazy loading for circular dependencies
+from kornia.augmentation.base import (
+    _AugmentationBase,
+    GeometricAugmentationBase2D,
+    MixAugmentationBase,
+    TensorWithTransformMat,
+)
+from kornia.constants import DataKey
+from kornia.geometry.bbox import transform_bbox
+from kornia.geometry.linalg import transform_points
+
+from .base import ParamItem
+
+
+def get_geometric_only_param(module: "CTN.ImageSequential", param: List[ParamItem]) -> List[ParamItem]:
+    named_modules = module.get_forward_sequence(param)
+
+    res = []
+    for (_, module), p in zip(named_modules, param):
+        if isinstance(module, (GeometricAugmentationBase2D,)):
+            res.append(p)
+    return res
+
+
+def make_input_only_sequential(module: "CTN.ImageSequential") -> Callable:
+    """Disable all other additional inputs (e.g. ) for ImageSequential."""
+    def f(*args, **kwargs):
+        if_return_trans = module.return_transform
+        if_return_label = module.return_label
+        module.return_transform = False
+        module.return_label = False
+        out = module(*args, **kwargs)
+        module.return_transform = if_return_trans
+        module.return_label = if_return_label
+        return out
+    return f
+
+
+class InputApplyInverse:
```

Comment:
Why do you need a class to hold just functions?

Reply:
It will be refactored later. I am thinking of making some classes to avoid the tons of if-else statements in https://github.com/kornia/kornia/blob/e7f3d3d9301efcdebbe044e04e1327e16c1063a7/kornia/augmentation/container/utils.py#L177-L184